dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1971,6 +1971,13 @@ void DeltaMergeStore::segmentMerge(DMContext & dm_context, const SegmentPtr & le
         right->info(),
         dm_context.min_version);
 
+    /// This segment may contain some rows that not belong to this segment range which is left by previous split operation.
+    /// And only saved data in this segment will be filtered by the segment range in the merge process,
+    /// unsaved data will be directly copied to the new segment.
+    /// So we flush here to make sure that all potential data left by previous split operation is saved.
+    left->flushCache(dm_context);
+    right->flushCache(dm_context);
```

Comment:
We need to check the result of flushCache, as it can be false.

Reply:
Fixed.